src/page/managers/LoginManager.ts

```diff
@@ -43,64 +51,72 @@ export default class LoginManager {
       return result;
   }
 
-  static async upsertUser(userData: Partial<UserData>): Promise<UserData> {
+  static async upsertUser(userData: Partial<UserData>, retry: number = 5): Promise<UserData> {
     logMethodCall("LoginManager.upsertUser", { userData });
+
+    if (retry === 0) {
+      throw new OneSignalError("Login: upsertUser failed: max retries reached");
+    }
+
     const appId = await MainHelper.getAppId();
     this.stripAliasesOtherThanExternalId(userData);
     const response = await RequestService.createUser({ appId }, userData);
     const result = response?.result;
     const status = response?.status;
 
-    if (status && status >= 200 && status < 300) {
+    if (status >= 200 && status < 300) {
       Log.info("Successfully created user", result);
-    } else if (status && status >= 400 && status < 500) {
+    } else if (status >= 400 && status < 500) {
       Log.error("Malformed request", result);
-    } else if (status && status >= 500) {
-      // retry indefinitely
+    } else if (status >= 500) {
```

Comment:
What status or result do we get on timeout? We should treat it like a 5xx so we can retry it.

Reply:
Typically a 408